src/libdrakvuf/libdrakvuf.h

```diff
@@ -223,6 +223,34 @@ struct drakvuf_trap {
     void *data;
 };
 
+
+////////////////////////////////////////////////////////////////////////////
+
+// IMHO these definitions must be placed within another file, named
+// libdrakvuf-windows.h or something similar
+
+// For get_previous_mode...
+typedef enum _MODE {
```

Comment:
Same as the point before: Being able to use it from the plugins adding more readability.

Take a look at this code from inside a plugin:
```
uint8_t previous_mode = -1 ;

drakvuf_get_previous_mode( drakvuf, info, &previous_mode );

if ( previous_mode == UserMode )
{
      uint32_t thread_id = 0 ;
```

IMHO this makes the code a lot cleaner